fdbserver/include/fdbserver/DataDistribution.actor.h

```diff
@@ -303,28 +320,41 @@ struct GetTopKMetricsRequest {
 	std::vector<KeyRange> keys;
 	Promise<GetTopKMetricsReply> reply; // topK storage metrics
 	double maxBytesReadPerKSecond = 0, minBytesReadPerKSecond = 0; // all returned shards won't exceed this read load
+	bool isTop; // Choose TopK OR BottomK
 
 	GetTopKMetricsRequest() {}
 	GetTopKMetricsRequest(std::vector<KeyRange> const& keys,
 	                      int topK = 1,
 	                      double maxBytesReadPerKSecond = std::numeric_limits<double>::max(),
-	                      double minBytesReadPerKSecond = 0)
+	                      double minBytesReadPerKSecond = 0,
+	                      bool isTop = true)
 	  : topK(topK), keys(keys), maxBytesReadPerKSecond(maxBytesReadPerKSecond),
-	    minBytesReadPerKSecond(minBytesReadPerKSecond) {}
+	    minBytesReadPerKSecond(minBytesReadPerKSecond), isTop(isTop) {}
 
 	// Return true if a.score > b.score, return the largest topK in keys
-	static bool compare(const GetTopKMetricsReply::KeyRangeStorageMetrics& a,
-	                    const GetTopKMetricsReply::KeyRangeStorageMetrics& b) {
-		return compareByReadDensity(a, b);
```

Comment:
I prefer to encapsulate the logic of handling `isTop` here rather than add extra functions
```
return isTop ? compareByReadDensityTop(...) ? compareByReadDensityBottom(...)
```